### ISSUE TITLE
Failed connection to rstp.

### ISSUE THREAD

**Issue by GMagician** [roflcoopter/viseron#756]
It could be a configuration issue but connection with vlc works, using another nvr and same path works, it seems that only viseron has this issue.

P.S. I added width, height, fps and other because I had ffprobe errors (even this works ok on other NVR)

here my config:
```
ffmpeg:
  camera:
    camera_1:
      name: Retro
      host: 192.168.0.114
      port: 554
      path: /0
      username: !secret username
      password: !secret password
      protocol: rtsp
      stream_format: rtsp
      width: 1944
      height: 2592
      fps: 25
      codec: h264
      audio_codec: null

darknet:
  object_detector:
    cameras:
      camera_1:
        fps: 1
        scan_on_motion_only: false
        labels:
          - label: person
            confidence: 0.75
            trigger_recorder: true

nvr:
  camera_1:
```

here the log
```
viseron  | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
viseron  | [s6-init] ensuring user provided files have correct perms...exited 0.
viseron  | [fix-attrs.d] applying ownership & permissions fixes...
viseron  | [fix-attrs.d] done.
viseron  | [cont-init.d] executing container initialization scripts...
viseron  | [cont-init.d] 10-adduser: executing... 
viseron  | [0;32m************************ UID/GID *************************[0m
viseron  | [0;32mUser uid:    1000[0m
viseron  | [0;32mUser gid:    1000[0m
viseron  | [0;32m************************** Done **************************[0m
viseron  | [cont-init.d] 10-adduser: exited 0.
viseron  | [cont-init.d] 20-gid-video-device: executing... 
viseron  | [cont-init.d] 20-gid-video-device: exited 0.
viseron  | [cont-init.d] 30-edgetpu-permission: executing... 
viseron  | [0;32m************** Setting EdgeTPU permissions ***************[0m
viseron  | [0;32mCoral Vendor IDs:
viseron  | "1a6e"
viseron  | "18d1"[0m
viseron  | [0;32mNo EdgeTPU USB device was found[0m
viseron  | [0;32mNo EdgeTPU PCI device was found[0m
viseron  | [0;32m************************** Done **************************[0m
viseron  | [cont-init.d] 30-edgetpu-permission: exited 0.
viseron  | [cont-init.d] 40-set-env-vars: executing... 
viseron  | [0;32m****** Checking for hardware acceleration platforms ******[0m
viseron  | [0;33mOpenCL cannot be used[0m
viseron  | [0;33mVA-API cannot be used[0m
viseron  | [0;33mCUDA cannot be used[0m
viseron  | [0;32m*********************** Done *****************************[0m
viseron  | [cont-init.d] 40-set-env-vars: exited 0.
viseron  | [cont-init.d] 50-check-if-rpi: executing... 
viseron  | [0;32m********** Checking if we are running on an RPi **********[0m
viseron  | [0;32mNot running on any supported RPi[0m
viseron  | [0;32m*********************** Done *****************************[0m
viseron  | [cont-init.d] 50-check-if-rpi: exited 0.
viseron  | [cont-init.d] 55-check-if-jetson: executing... 
viseron  | [0;32m****** Checking if we are running on a Jetson Board ******[0m
viseron  | [0;32mNot running on any supported Jetson board[0m
viseron  | [0;32m*********************** Done *****************************[0m
viseron  | [cont-init.d] 55-check-if-jetson: exited 0.
viseron  | [cont-init.d] 60-ffmpeg-path: executing... 
viseron  | [0;32m****************** Getting FFmpeg path *******************[0m
viseron  | [0;32mFFmpeg path: /home/abc/bin/ffmpeg[0m
viseron  | [0;32m*********************** Done *****************************[0m
viseron  | [cont-init.d] 60-ffmpeg-path: exited 0.
viseron  | [cont-init.d] 70-gstreamer-path: executing... 
viseron  | [0;32m***************** Getting GStreamer path *****************[0m
viseron  | [0;32mGStreamer path: /usr/bin/gst-launch-1.0[0m
viseron  | [0;32m*********************** Done *****************************[0m
viseron  | [cont-init.d] 70-gstreamer-path: exited 0.
viseron  | [cont-init.d] done.
viseron  | [services.d] starting services
viseron  | [services.d] done.
viseron  | [32m[2024-05-19 22:42:33] [INFO    ] [viseron.core] - -------------------------------------------[0m
viseron  | [32m[2024-05-19 22:42:33] [INFO    ] [viseron.core] - Initializing Viseron[0m
viseron  | [32m[2024-05-19 22:42:33] [INFO    ] [viseron.components] - Setting up component data_stream[0m
viseron  | [32m[2024-05-19 22:42:33] [INFO    ] [viseron.components] - Setup of component data_stream took 0.0 seconds[0m
viseron  | [32m[2024-05-19 22:42:33] [INFO    ] [viseron.components] - Setting up component webserver[0m
viseron  | [32m[2024-05-19 22:42:33] [INFO    ] [viseron.components] - Setup of component webserver took 0.0 seconds[0m
viseron  | [32m[2024-05-19 22:42:33] [INFO    ] [viseron.components] - Setting up component ffmpeg[0m
viseron  | [32m[2024-05-19 22:42:33] [INFO    ] [viseron.components] - Setting up component darknet[0m
viseron  | [32m[2024-05-19 22:42:33] [INFO    ] [viseron.components] - Setting up component nvr[0m
viseron  | [32m[2024-05-19 22:42:33] [INFO    ] [viseron.components] - Setup of component nvr took 0.0 seconds[0m
viseron  | [32m[2024-05-19 22:42:33] [INFO    ] [viseron.components] - Setup of component ffmpeg took 0.0 seconds[0m
viseron  | [32m[2024-05-19 22:42:33] [INFO    ] [viseron.components] - Setup of component darknet took 0.1 seconds[0m
viseron  | [32m[2024-05-19 22:42:34] [INFO    ] [viseron.components] - Setting up domain camera for component ffmpeg with identifier camera_1[0m
viseron  | [32m[2024-05-19 22:42:34] [INFO    ] [viseron.components] - Setting up domain object_detector for component darknet with identifier camera_1[0m
viseron  | [32m[2024-05-19 22:42:34] [INFO    ] [viseron.components] - Setting up domain nvr for component nvr with identifier camera_1[0m
viseron  | [32m[2024-05-19 22:42:34] [INFO    ] [viseron.components] - Setup of domain camera for component ffmpeg with identifier camera_1 took 0.0 seconds[0m
viseron  | [32m[2024-05-19 22:42:34] [INFO    ] [viseron.components] - Setup of domain object_detector for component darknet with identifier camera_1 took 0.0 seconds[0m
viseron  | [32m[2024-05-19 22:42:34] [INFO    ] [viseron.components.nvr.nvr.camera_1] - Motion detector is disabled[0m
viseron  | [32m[2024-05-19 22:42:34] [INFO    ] [viseron.components.nvr.nvr.camera_1] - NVR for camera Retro initialized[0m
viseron  | [32m[2024-05-19 22:42:34] [INFO    ] [viseron.components] - Setup of domain nvr for component nvr with identifier camera_1 took 0.0 seconds[0m
viseron  | [32m[2024-05-19 22:42:34] [INFO    ] [viseron.core] - Viseron initialized in 0.6 seconds[0m
viseron  | [31m[2024-05-19 22:42:34] [ERROR   ] [viseron.components.ffmpeg.stream.camera_1] - [tcp @ 0x55d5f62e1cc0] Connection to tcp://:554?timeout=5000000 failed: Connection refused[0m
viseron  | [31m[2024-05-19 22:42:34] [ERROR   ] [viseron.components.ffmpeg.stream.camera_1] - rtsp://*****:*****@192.168.0.114:554/0: Connection refused[0m
viseron  | [31m[2024-05-19 22:42:34] [ERROR   ] [viseron.components.ffmpeg.camera.camera_1] - Frame reader process has exited[0m
viseron  | [31m[2024-05-19 22:42:39] [ERROR   ] [viseron.components.ffmpeg.camera.camera_1] - Restarting frame pipe[0m
viseron  | [31m[2024-05-19 22:42:39] [ERROR   ] [viseron.components.ffmpeg.stream.camera_1] - [tcp @ 0x556331154cc0] Connection to tcp://:554?timeout=5000000 failed: Connection refused[0m
viseron  | [31m[2024-05-19 22:42:39] [ERROR   ] [viseron.components.ffmpeg.stream.camera_1] - rtsp://*****:*****@192.168.0.114:554/0: Connection refused[0m
viseron  | [31m[2024-05-19 22:42:39] [ERROR   ] [viseron.components.ffmpeg.camera.camera_1] - Frame reader process has exited[0m
viseron  | [31m[2024-05-19 22:42:44] [ERROR   ] [viseron.components.ffmpeg.camera.camera_1] - Restarting frame pipe[0m
viseron  | [31m[2024-05-19 22:42:44] [ERROR   ] [viseron.components.ffmpeg.stream.camera_1] - [tcp @ 0x562c69335cc0] Connection to tcp://:554?timeout=5000000 failed: Connection refused[0m
viseron  | [31m[2024-05-19 22:42:44] [ERROR   ] [viseron.components.ffmpeg.stream.camera_1] - rtsp://*****:*****@192.168.0.114:554/0: Connection refused[0m
viseron  | [31m[2024-05-19 22:42:44] [ERROR   ] [viseron.components.ffmpeg.camera.camera_1] - Frame reader process has exited[0m
viseron  | [31m[2024-05-19 22:42:49] [ERROR   ] [viseron.components.ffmpeg.camera.camera_1] - Restarting frame pipe[0m
viseron  | [31m[2024-05-19 22:42:49] [ERROR   ] [viseron.components.ffmpeg.stream.camera_1] - [tcp @ 0x5650725f5cc0] Connection to tcp://:554?timeout=5000000 failed: Connection refused[0m
viseron  | [31m[2024-05-19 22:42:49] [ERROR   ] [viseron.components.ffmpeg.stream.camera_1] - rtsp://*****:*****@192.168.0.114:554/0: Connection refused[0m
viseron  | [31m[2024-05-19 22:42:49] [ERROR   ] [viseron.components.ffmpeg.camera.camera_1] - Frame reader process has exited[0m
viseron  | [31m[2024-05-19 22:42:54] [ERROR   ] [viseron.components.ffmpeg.camera.camera_1] - Restarting frame pipe[0m
viseron  | [31m[2024-05-19 22:42:54] [ERROR   ] [viseron.components.ffmpeg.stream.camera_1] - [tcp @ 0x55c13c603cc0] Connection to tcp://:554?timeout=5000000 failed: Connection refused[0m
viseron  | [31m[2024-05-19 22:42:54] [ERROR   ] [viseron.components.ffmpeg.stream.camera_1] - rtsp://*****:*****@192.168.0.114:554/0: Connection refused[0m
viseron  | [31m[2024-05-19 22:42:54] [ERROR   ] [viseron.components.ffmpeg.camera.camera_1] - Frame reader process has exited[0m
```

**Comment by GMagician** [comment]
Addendum: I double checked username and password in "secret" and they are correct

**Comment by GMagician** [comment]
Because of #755 I manually removed admin and password...

**Comment by GMagician** [comment]
This is because of chars used by password!

**Comment by roflcoopter** [comment]
So you hade special characters in your password?
What were those special characters? Need to find a way to fix that

**Comment by GMagician** [comment]
Thanks for reply.

Meantime I changed my password to try to solve.
@ is for sure a problem but it seems also something else in this list `&;,*`. With my new password, if I launch ffprobe I still get errors regarding `;`. I discovered that passing password in between double quotes it runs..eg `ffprobe rtsp://user:"password"@ippath`

**Comment by roflcoopter** [comment]
Done some changes to escape special characters in the password, ran some tests and it seems to work fine now!
Will push it to the next beta soon

**Comment by roflcoopter** [comment]
This is now released in `3.0.0b7`
